### PR TITLE
(#158) Implement distinguishing between different kinds of bindings in BDB 

### PR DESCRIPTION
**Pull request by herrhotzenplotz**
See #158 

Also adds a TODO.

Please let me know if you have any issues with this PR.

### CODE REVIEW

File: src/bdb.c [hunk 1]
```diff
@@ -276,24 +324,29 @@ Bdb_Err bdb_run_command(Bdb_State *state, String_View command_word, String_View
         String_View where_sv = sv_chop_by_delim(&arguments, ' ');
         String_View count_sv = arguments;
 
-        Inst_Addr where = 0;
-        if (bdb_parse_label_or_addr(state, where_sv, &where) == BDB_FAIL)
+        // TODO: Rethink the types in the `examine` command of bdb
```
Comment (rexim) [inline src/bdb.c:327]: What do you mean by "rethink"? What's the problem here?

**Review by rexim — CHANGES_REQUESTED** [review]

**Comment by herrhotzenplotz** [comment]
On 1/30/21 8:42 AM, Alexey Kutepov wrote:
> @rexim requested changes on this pull request.
> 
> 
> 
>> @@ -276,24 +324,29 @@ Bdb_Err bdb_run_command(Bdb_State *state, String_View command_word, String_View
>           String_View where_sv = sv_chop_by_delim(&arguments, ' ');
>           String_View count_sv = arguments;
>   
> -        Inst_Addr where = 0;
> -        if (bdb_parse_label_or_addr(state, where_sv, &where) == BDB_FAIL)
> +        // TODO: Rethink the types in the `examine` command of bdb
> 
> What do you mean by "rethink"? What's the problem here?
> 


That was merely me, working with the unintuitive types. I may refactor 
this later, because juggling around with the `Word` union was really 
awkward. However, we may just remove this TODO for now if it is 
distracting or inappropriate.

-- 
Sent from Thunderbird@Triton / FreeBSD 13-ALPHA2

Please remember: https://useplaintext.email/#etiquette
Do NOT send me Microsoft Office documents. I will and cannot open them.

Nico Sonack <nsonack@outlook.com>

**Comment by rexim** [comment]
@herrhotzenplotz yeah, it sounds too vague. I'm just afraid that after some time we won't be able to recall what it was about. Neither it is a very important issue imo.

**Review by rexim — APPROVED** [review]

**Comment by rexim** [comment]
@herrhotzenplotz thank you for your contribution!

**Pull request merged**